CMakeLists.txt

```diff
@@ -47,6 +47,8 @@ option(QL_ENABLE_PARALLEL_UNIT_TEST_RUNNER "Enable the parallel unit test runner
 option(QL_ENABLE_SESSIONS "Singletons return different instances for different sessions" OFF)
 option(QL_ENABLE_THREAD_SAFE_OBSERVER_PATTERN "Enable the thread-safe observer pattern" OFF)
 option(QL_ENABLE_TRACING "Tracing messages should be allowed" OFF)
+option(QL_ENABLE_DEFAULT_WARNING_LEVEL "Enable the default warning level to pass the ci pipeline" ON)
```

Comment:
I'd keep this off by default and turn it on in the CI build.  I guess this means it wouldn't be a DEFAULT_WARNING_LEVEL? Maybe a STRICT_WARNING_LEVEL?

Reply:
That's probably the same discussion as above?! 🤣

I think in some sense it is the `QL_DEFAULT_WARNING_LEVEL` as a release guaranties to be warning free using this option. That's what I would call a default (and why I enabled it by default).

Reply:
Probably the same.  Ok, fine, I guess all you cmake users like your warnings anyway 😄